src/common/gitservice/blame.go

```diff
@@ -19,16 +21,50 @@ type GitBlame struct {
 	GitUserEmail  string
 }
 
-func NewGitBlame(filePath string, lines structure.Lines, blameResult *git.BlameResult, gitOrg string, gitRepository string, userEmail string) *GitBlame {
-	gitBlame := GitBlame{GitOrg: gitOrg, GitRepository: gitRepository, BlamesByLine: map[int]*git.Line{}, FilePath: filePath, GitUserEmail: userEmail}
+func GetPreviousBlameResult(g *GitService, filePath string) (*git.BlameResult, *object.Commit) {
+	if g.repository == nil {
+		return nil, nil
+	}
+	ref, err := g.repository.Head()
+	if err != nil {
+		return nil, nil
+	}
+	commit, err := g.repository.CommitObject(ref.Hash())
+	if err != nil {
+		return nil, nil
+	}
+	parentIter := commit.Parents()
+	previousCommit, err := parentIter.Next()
+	if err != nil {
+		return nil, nil
+	}
+
+	var previousBlameResult *git.BlameResult
+	result, _ := g.PreviousBlameByFile.Load(filePath)
+	previousBlameResult = result.(*git.BlameResult)
+	return previousBlameResult, previousCommit
+}
+
+func NewGitBlame(relativeFilePath string, filePath string, lines structure.Lines, blameResult *git.BlameResult, g *GitService) *GitBlame {
+	gitBlame := GitBlame{GitOrg: g.organization, GitRepository: g.repoName, BlamesByLine: map[int]*git.Line{}, FilePath: relativeFilePath, GitUserEmail: g.currentUserEmail}
 	startLine := lines.Start - 1 // the lines in blameResult.Lines start from zero while the lines range start from 1
 	endLine := lines.End - 1
+	previousBlameResult, previousCommit := GetPreviousBlameResult(g, filePath)
+
 	for line := startLine; line <= endLine; line++ {
 		if line >= len(blameResult.Lines) {
-			logger.Warning(fmt.Sprintf("Index out of bound on parsed file %s", filePath))
+			logger.Warning(fmt.Sprintf("Index out of bound on parsed file %s", relativeFilePath))
 			return &gitBlame
 		}
 		gitBlame.BlamesByLine[line+1] = blameResult.Lines[line]
+
+		// Check if the line has been removed in the current state of the file
+		if previousBlameResult != nil && len(previousBlameResult.Lines) > len(blameResult.Lines) {
```

Comment:
not sure about this check `len(previousBlameResult.Lines) > len(blameResult.Lines)` will this always be true or are there cases where is `<=` but still the specific line was removed?